libs/hdf-converters/src/ckl-mapper/checklist-mapper.ts

```diff
@@ -296,10 +296,28 @@ function getHdfSpecificDataAttribute(
 }
 
 /**
- * ChecklistResults is a wrapper for ChecklistMapper using the intakeType
- *  default returns a single hdf object without any modifications
- *  split returns multiple hdf object based on number of iSTIG objects in checklist
- *  wrapper returns a single hdf object with an additional profile created using file name as profile name and adds parent_profile key to each mapped profile
+ * The `ChecklistResults` class extends the `ChecklistJsonixConverter` and is responsible for converting
+ * checklist data between different formats (XML CKL, HDF JSON).
+ *
+ * @extends ChecklistJsonixConverter
+ *
+ * @property {string | ExecJSON.Execution} data - The input data, which can be a string of XML data or an HDF JSON execution object.
+ * @property {Checklist} jsonixData - The JSON representation of the checklist data.
+ * @property {ChecklistObject} checklistObject - The intermediate object representation of the checklist data.
+ * @property {boolean} withRaw - A flag indicating whether to include raw data in the output.
+ *
+ * @constructor
```

Comment:
why are we defining the constructor parameters both here and on top of the constructor?